### ISSUE TITLE
Passing reference type parameters to Patient resource throws error

### ISSUE THREAD

**Issue by FSmithYDH** [Asymmetrik/node-fhir-server-mongo#28]
### Do you want to request a feature, improve documentation, or ask a question?

N/A
### Are you reporting a bug?

Yes

**What is the current behavior?**

Error thrown: Invalid parameter: link

**What is the expected behavior?**
Patient resources that meet the criteria set in the search parameter are returned.

**What are the steps to reproduce?**
Send the following request to the server:

```
GET /3_0_1/Patient?link=400 HTTP/1.1
Host: 127.0.0.1:3000
Content-Type: application/fhir+json
Cache-Control: no-cache
```

**What OS are you using and what version of node.js, @asymmetrik/node-fhir-server-core, and @asymmetrik/node-fhir-server-mongo are you running?**
OS version: Windows 10 - 64-bit, version 1607
node.js version: 10.11.0
node-fhir-server-core version: 1.2.0
node-fhir-server-mongo version: 1.1.0

**Comment by Robert-W** [comment]
Thanks @FSmithYDH, this looks like a bug in the core library. We'll take a look at this. The core library rejects any parameters it does not know about it and there is probably a misconfiguration there.

**Comment by FSmithYDH** [comment]
Thanks @Robert-W , don't know if it's worth opening another issue for this but URI type parameters throw an error as well.

**Comment by Robert-W** [comment]
We can take care of that at the same time. I checked core and the parameters are configured correctly, but the parameter parser is being very restrictive, so we just need to update it to allow reference and uri types through. Should have core updated soon.

**Comment by Robert-W** [comment]
Hey @FSmithYDH, this is no fixed in core, we have not made the update to the master branch since we are still testing some things in our develop branch. If you want to try it out, you can update the dependencies in your package.json to look like this:

```
"@asymmetrik/node-fhir-server-core": "git+https://github.com/Asymmetrik/node-fhir-server-core.git#develop",
```

**Comment by FSmithYDH** [comment]
Thanks @Robert-W, going to be a pain and throw another type in as well: composite types like the 'Relationship' parameter for DocumentReference.

**Comment by Robert-W** [comment]
Hey @FSmithYDH, not a pain at all. I really appreciate you helping us find all these cases. We'll get this updated as soon as we can.

We are building in parallel a new rest sanitizer and we are going to specifically add these as test cases (we are actually building a suite of tools including this to help others build these servers and manage testing/features more easily but those probably won't be ready til after DevDays in Amsterdam). So once it's ready, we won't be blocking any more legitimate parameters.
